File: app/src/main/java/net/twisterrob/android/view/color/swatches/pixel/drawer/DrawerUtils.java
package net.twisterrob.android.view.color.swatches.pixel.drawer;

public class DrawerUtils {
    public static void drawBorder(BitmapDrawer drawer, int color, int width) {
        int h = drawer.h;
        int w = drawer.w;
        int[] bitmap = drawer.bitmap;
        for (int y = 0; y < h; y++) {
            for (int x = 0; x < width; x++) {
                bitmap[y * w + x] = color;
                bitmap[y * w + (w - x - 1)] = color;
            }
        }
        for (int x = 0; x < w; x++) {
            for (int y = 0; y < width; y++) {
                bitmap[y * w + x] = color;
                bitmap[(h - y - 1) * w + x] = color;
            }
        }
    }
}

File: feature/base/src/main/kotlin/net/twisterrob/android/view/color/swatches/pixel/drawer/DrawerUtils.kt
@file:JvmName("DrawerUtils")

package net.twisterrob.android.view.color.swatches.pixel.drawer

fun drawBorder(drawer: BitmapDrawer, color: Int, width: Int) {
	val h = drawer.h
	val w = drawer.w
	val bitmap = drawer.bitmap
	for (y in 0 until h) {
		for (x in 0 until width) {
			bitmap[y * w + x] = color
			bitmap[y * w + (w - x - 1)] = color
		}
	}
	for (x in 0 until w) {
		for (y in 0 until width) {
			bitmap[y * w + x] = color
			bitmap[(h - y - 1) * w + x] = color
		}
	}
}
